plugins/aja/aja-card-manager.cpp

```diff
@@ -0,0 +1,618 @@
+			       const std::string &owner)
+{
+	bool acquired = false;
+
+	if (ChannelReady(chan, owner)) {
+		const std::lock_guard<std::mutex> lock(mMutex);
+
+		if (mChannelPwnz.find(owner) != mChannelPwnz.end()) {
+			if (mChannelPwnz[owner] &
+			    (1 << static_cast<int32_t>(chan))) {
+				acquired = true;
+			} else {
+				mChannelPwnz[owner] |=
+					(1 << static_cast<int32_t>(chan));
+				acquired = true;
+			}
+		} else {
+			mChannelPwnz[owner] |=
+				(1 << static_cast<int32_t>(chan));
+			acquired = true;
+		}
+
+		// Acquire interrupt handles
+		if (acquired && mCard) {
+			if (mode == NTV2_MODE_CAPTURE) {
+				mCard->EnableInputInterrupt(chan);
+				mCard->SubscribeInputVerticalEvent(chan);
+			} else if (mode == NTV2_MODE_DISPLAY) {
+				mCard->EnableOutputInterrupt(chan);
+				mCard->SubscribeOutputVerticalEvent(chan);
+			}
+		}
+	}
+
+	return acquired;
+}
+
+bool CardEntry::ReleaseChannel(NTV2Channel chan, NTV2Mode mode,
+			       const std::string &owner)
+{
+	const std::lock_guard<std::mutex> lock(mMutex);
+
+	for (const auto &pwn : mChannelPwnz) {
+		if (pwn.first == owner) {
+			if (mChannelPwnz[owner] &
+			    (1 << static_cast<int32_t>(chan))) {
+				mChannelPwnz[owner] ^=
+					(1 << static_cast<int32_t>(chan));
+
+				// Release interrupt handles
+				if (mCard) {
+					if (mode == NTV2_MODE_CAPTURE) {
+						mCard->DisableInputInterrupt(
+							chan);
+						mCard->UnsubscribeInputVerticalEvent(
+							chan);
+					} else if (mode == NTV2_MODE_DISPLAY) {
+						mCard->DisableOutputInterrupt(
+							chan);
+						mCard->UnsubscribeOutputVerticalEvent(
+							chan);
+					}
+				}
+
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
+bool CardEntry::InputSelectionReady(IOSelection io, NTV2DeviceID id,
+				    const std::string &owner) const
+{
+	// uncomment if we add device-specific logic
+	UNUSED_PARAMETER(id);
+
+	NTV2InputSourceSet inputSources;
+	aja::IOSelectionToInputSources(io, inputSources);
+
+	if (inputSources.size() > 0) {
+		size_t channelsReady = 0;
+
+		for (auto &&src : inputSources) {
+			auto channel = NTV2InputSourceToChannel(src);
+			if (ChannelReady(channel, owner))
+				channelsReady++;
+		}
+
+		if (channelsReady == inputSources.size())
+			return true;
+	}
+
+	return false;
+}
+
+bool CardEntry::OutputSelectionReady(IOSelection io, NTV2DeviceID id,
+				     const std::string &owner) const
+{
+	// Handle checking special case outputs --
+	// HDMI Monitor uses framestore 4
+	if (aja::CardCanDoHDMIMonitorOutput(id) &&
+	    io == IOSelection::HDMIMonitorOut) {
+		NTV2Channel hdmiMonChannel = NTV2_CHANNEL4;
+		return ChannelReady(hdmiMonChannel, owner);
+	}
+	// SDI Monitor on io4K/io4K+/etc. uses framestore 4
+	else if (aja::CardCanDoSDIMonitorOutput(id) &&
+		 io == IOSelection::SDI5) {
+		NTV2Channel sdiMonChannel = NTV2_CHANNEL4;
+		return ChannelReady(sdiMonChannel, owner);
+	}
+	// Check readiness of all other channels
+	else {
```

Comment:
Should I move the comments so that they appear before the entire if-else block?
For example:
```
	/* Handle checking special case outputs before all other outputs.
	 * 1. HDMI Monitor uses framestore 4
	 * 2. SDI Monitor on Io 4K/Io 4K Plus, etc. uses framestore 4.
	 * 3. Everything else...
	 */
	if (aja::CardCanDoHDMIMonitorOutput(id) &&
	    io == IOSelection::HDMIMonitorOut) {
		NTV2Channel hdmiMonChannel = NTV2_CHANNEL4;
		return ChannelReady(hdmiMonChannel, owner);
	} else if (aja::CardCanDoSDIMonitorOutput(id) &&
		   io == IOSelection::SDI5) {
    ...
```

Reply:
Either placing them before the entire statement, as you have demonstrated, or placing an appropriate comment within each block would be fine.  Alternatively, placing a detailed comment block before the entire statement and also placing short single line summary comments in each block would also be fine.

If the comments become too unwieldy or get in the way of reading the code, I'm sure someone else will mention it during the review process.